client/transport.go

```diff
@@ -0,0 +1,54 @@
+package client
+
+import (
+	"crypto/tls"
+	"errors"
+	"net/http"
+)
+
+var errTLSConfigUnavailable = errors.New("TLSConfig unavailable")
+
+// transportFunc allows us to inject a mock transport for testing. We define it
+// here so we can detect the tlsconfig and return nil for only this type.
+type transportFunc func(*http.Request) (*http.Response, error)
+
+func (tf transportFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return tf(req)
+}
+
+// resolveTLSConfig attempts to resolve the tls configuration from the
+// RoundTripper.
+func resolveTLSConfig(transport http.RoundTripper) (*tls.Config, error) {
+	if tr, ok := transport.(*http.Transport); ok {
+		return tr.TLSClientConfig, nil
+	}
+
+	switch tr := transport.(type) {
+	case *http.Transport:
+		return tr.TLSClientConfig, nil
+	case transportFunc:
+		return nil, nil // detect this type for testing.
+	default:
+		return nil, errTLSConfigUnavailable
+	}
+}
+
+// resolveScheme detects a tls config on the transport and returns the
+// appropriate http scheme.
+//
+// TODO(stevvooe): This isn't really the right way to write clients in Go.
+// There should be a "base" url, that includes the scheme, and the client
+// should build urls from that. We leave this trash around to keep things
```

Comment:
If we keep the code standards high, the tone will reflect that quality. Adjusting the tone to fit some arbitrary standards of polite society only serves to continue the problem. Put another way, if you want to fix the tone, let's address the root cause rather than prettying up a turd.

I've fixed it but I don't think it is worth the time or the extra round trip to jenkins to address this.